app/src/main/java/org/mozilla/focus/browser/BrowserNavigationOverlay.kt

```diff
@@ -68,6 +70,16 @@ class BrowserNavigationOverlay @JvmOverloads constructor(
      */
     private var uiLifecycleCancelJob: Job
 
+    /**
+     * Setting the openContextMenu function in the HomeTileAdapter is fragile
+     * since we need to use Delegates.observable to update the function. However,
+     * the function will be updated by then because it's only used when a user
+     * needs to long press on a tile, after all the views have been created.
+     */
+    var openHomeTileContextMenu: (() -> Unit) by Delegates.observable(fun () { }) { _, _, newValue ->
+        setOpenContextMenuFunctionInAdapter(newValue)
```

Comment:
nit: call the code directly instead of creating a new function. Adding unnecessary functions pollutes auto-complete and makes it hard to figure out what a class does.